fix: Missing keyboard accessibility for AI toolbar button functionality

syuzanvs opened the pull request:
- Updated the keyboard event listener in the `AiAgentUI` class to improve the handling of the AI Agent command execution.
- The shortcut for executing the command has been refined to ensure better responsiveness and user experience.
- Added a check for the `event` parameter in the `keydown` event listener to ensure that the command is executed only when the specified key combination (Ctrl + /) is pressed.

Closes #61

syuzanvs committed:
fix: Missing keyboard accessibility for AI toolbar button functionality

- Updated the keyboard event listener in the `AiAgentUI` class to improve the handling of the AI Agent command execution.
- The shortcut for executing the command has been refined to ensure better responsiveness and user experience.
- Added a check for the `event` parameter in the `keydown` event listener to ensure that the command is executed only when the specified key combination (Ctrl + /) is pressed.

Closes #61

jjroelofs reviewed (changes requested):
## Wrong key-shortcut implementation

The issue specified `ctrl/CMD + /` is the required keystroke combination, this means `ctrl` on Windows and `CMD` on MacOS. This is the convention in CKEditor and the OS difference is handled automatically by the CK5 API:

https://ckeditor.com/docs/ckeditor5/latest/framework/tutorials/crash-course/keystrokes.html

By using a custom keystrokes implementation we end up with a behavior not conforming to the set requirements.

## Add, not replace a shortcut

`ctrl/cmd + /` and `/` are two different features and we need to keep both in the accessibility panel. The singular `/` is the keystroke to enter a AI command in an empty element, `ctrl/CMD+/ `is a more specialized combination to force a slash command inside a pre-existing text. Im not sure what description text to use, pls optimize this description text in the Accessibility Help panel with your LLM of choice (ask the LLM to be an expert microcopy writer and UX researcher).

## Grouping

We can't have 3 AI Agent commands spread all over the panel, they should be grouped in an AI Agent group.

syuzanvs commented:
> ## Wrong key-shortcut implementation
> The issue specified `ctrl/CMD + /` is the required keystroke combination, this means `ctrl` on Windows and `CMD` on MacOS. This is the convention in CKEditor and the OS difference is handled automatically by the CK5 API:
> 
> https://ckeditor.com/docs/ckeditor5/latest/framework/tutorials/crash-course/keystrokes.html
> 
> By using a custom keystrokes implementation we end up with a behavior not conforming to the set requirements.
> 
> ## Add, not replace a shortcut
> `ctrl/cmd + /` and `/` are two different features and we need to keep both in the accessibility panel. The singular `/` is the keystroke to enter a AI command in an empty element, `ctrl/CMD+/ `is a more specialized combination to force a slash command inside a pre-existing text. Im not sure what description text to use, pls optimize this description text in the Accessibility Help panel with your LLM of choice (ask the LLM to be an expert microcopy writer and UX researcher).
> 
> ## Grouping
> We can't have 3 AI Agent commands spread all over the panel, they should be grouped in an AI Agent group.

@jjroelofs
The "keystrokes " didn't work for exactly command `ctrl/CMD + /`. I was not sure why, and that's the reason I got it done differently. But will recheck again to see what's the reason of the issue

syuzanvs committed:
Merge branch '1.x' of github.com:dxpr/ckeditor5-ai-agent into syuzan/1.x/#61-missing-keyboard-accessibility-for-ai-toolbar-button-functionality

syuzanvs committed:
fix: Missing keyboard accessibility for AI toolbar button functionality

- Added keystroke information for AI Agent commands under the 'navigation' category.
- Included specific keystrokes for both Mac and Windows environments:
    - `/` for opening the AI Command Menu in an empty field.
    - `Cmd + /` or `Ctrl + /` for forcing insert slash command within existing text.
    - `Cmd + Backspace` or `Ctrl + Backspace` for canceling AI generation.
- Added event listener for `keydown` to execute AI Agent command:
    - Executes the command when `Ctrl + /` or `Cmd + /` is pressed.

Closes #61

syuzanvs committed:
chore: Update pre-commit hook to include translations collection

- Updated the `pre-commit` script in `package.json` to run `lint-staged` and `translations:collect`.

Closes #61

syuzanvs commented:
@jjroelofs  
https://ckeditor.com/docs/ckeditor5/latest/framework/tutorials/crash-course/keystrokes.html
regarding you comment for this documentation, with this method it supports `Ctrl+Alt+H` but it doesn't support `Ctrl+/` . 
With my new change now our code should support MacOS but I'll need to ask you to test it

jjroelofs reviewed (changes requested):
I updated the keyboard shortcut descriptions and it is now looking good. However, the cancelation shortcut is not working for me. Click the cancel button works, using the keyboard shortcut does not. Not sure if it is a recent regression or we just didn't test it well before.

syuzanvs committed:
Fix conflict

syuzanvs commented:
@jjroelofs the cancelation works for me, was not able to reproduce an issue with that. 
If you can pls review again and let me know if you get it and how to reproduce. We can have a separate task for it if you are able to get the issue

jjroelofs committed:
Merge branch '1.x' into syuzan/1.x/#61-missing-keyboard-accessibility-for-ai-toolbar-button-functionality

jjroelofs reviewed (approved):
Great! And I cannot reproduce the failing cancel button anymore. Will create a new issue if it comes back :)
